### PR TITLE
Remove dependency on internal IP to detect routing loops

### PR DESCRIPTION
**Pull request by tgraf**
So far, the datapath has applied a masked version of the internal IP to
determine whehther an IP is potentially headed to a local endpoint or node by
associating the destination IP with the local allocation range. This is pretty
fragile and will break as we stop allocating from individual allocation CIDRs.

It was already fragile as certain IPs were blacklisted and would thus had to be
excluded as well.

Change the logic to use the ipcache information to determine whether we know
about the IP address or not and drop any packet that has been routed into
cilium_host for which no ipcache entry exists.

This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8030)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
Adjusted commit messages

**Comment by tgraf** [comment]
test-me-please

File: bpf/lib/common.h [hunk 1]
```diff
@@ -258,7 +258,7 @@ enum {
 #define DROP_NO_TUNNEL_OPT_	-148 /* unused */
 #define DROP_INVALID_GENEVE_	-149 /* unused */
 #define DROP_UNKNOWN_TARGET	-150
-#define DROP_NON_LOCAL		-151
+#define DROP_UNROUTABLE		-151
```
Comment (raybejjani) [inline bpf/lib/common.h:261]: an extra tab snuck in, it seems

**Review by raybejjani — APPROVED** [review]
LGTM janitor

File: bpf/lib/common.h (hunk 1 above)
Comment (tgraf) [reply bpf/lib/common.h:261]: The tab is correct in the editor, it aligns properly to the other numbers. It's different tab settings on GH.

**Comment by tgraf** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Pull request merged**